binary_gentoo/internal/cli/tree_diff.py

```diff
@@ -3,11 +3,26 @@
 
 import filecmp
 import os
+import re
 import sys
 from argparse import ArgumentParser
 
-from ..reporter import exception_reporting
 from ._parser import add_version_argument_to
+from ..reporter import exception_reporting
+
+keywords_pattern = re.compile('KEYWORDS="(?P<keywords>.*?)"')
```

Comment:
I made it non-greedy because the match is done over the whole ebuild content instead of looping over it line by line and matching per line (one less loop to manage..), and this way it is certain that it will stop matching as soon as it finds the next quote.

Reply:
For stopping at the first quote `"[^"]*"` would work well, even in multi-line context. Non-greedy quantifiers are an advanced feature of regex and I would personally want to reserve its use to where really needed, which does not seem the case here. I have not compared the performance of the two, my focus is simplicity here at the moment.

Reply:
Yes indeed, that would work too so I'll change.